src/transformStateWithClones.js

```
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let ArchiveOfState = [];
+  let currentState = { ...state };
+
+  for (let action of actions) {
+    let stateCopy = { ...currentState };
+
+    switch (action.type) {
+      case ('clear'):
+        stateCopy = {};
+        break;
+
+      case ('addProperties'):
+        Object.assign(stateCopy, action.extraData);
+        break;
+
+      case ('removeProperties'):
+        for (const key of action.keysToRemove) {
+          delete stateCopy[key];
+        }
+        break;
+
+    default:
+       return "unknown action types"
```

The `default` case currently returns a string ("unknown action types"). According to the requirements, the function should throw an error or handle unknown action types appropriately. Returning a string here will cause the function to return a string instead of an array, which is not the intended behavior. Consider replacing this with `throw new Error('Unknown action type')`. This violates checklist item: 'The switch statement must always have a default case for error handling (e.g., throw an error or handle unknown action types appropriately).'